### PR TITLE
Don't commit assets

### LINKED ISSUE #192: Use the same way to handle go-bindata as other projects
Looks like now we have to commit `asset.go`, but it could be avoided as done in other projects.

### PR DESCRIPTION
**Pull request by smacker**
fix #192

Now it works the same as any other apps project.
I also replaced bindata with esc.
`static_files_dev.go` & `static_files.go` are 100% copy-paste from lookout-web.

Makefile still looks a bit strange but changing it is out of the scope.

### CODE REVIEW

**Comment by carlosms** [comment]
It's tested this locally, and for me `make build` & `make serve` fail if the dir `./server/asset` does not exist.
When the dir exists, `make serve` works, but `make packages` creates a bin serving the 501 errors.

**Comment by smacker** [comment]
thanks for testing! I'll re-test with a clean state.

**Comment by smacker** [comment]
good idea! 👍

**Comment by smacker** [comment]
1. I added `mkdir -p` to make sure directory always exists
2. I can't reproduce the problem with `make packages`. This command and `make build` on clean repository generate correct binary with assets included.
3. I added simple integration test with check for correct content in `/` path
4. Had to rebase because `asset.go` was changed in master

**Comment by dpordomingo** [comment]
pretty good idea adding integration tests ! :tada:

File: Makefile [hunk 1]
```diff
@@ -2,31 +2,42 @@
...
-$(MAKEFILE):
-	git clone --quiet --depth 1 -b $(CI_BRANCH) $(CI_REPOSITORY) $(CI_PATH);
--include $(MAKEFILE)
+WITH_STATIC_TAG := with_static
+
+# Override Makefile.main defaults for arguments to be used in `go` commands.
+GO_BUILD_ARGS := -ldflags "$(LD_FLAGS)" -tags "$(WITH_STATIC_TAG)"
+
+# Environment and arguments to use in `go run` calls.
+GO_RUN_ARGS += -tags "$(WITH_STATIC_TAG)"
```
[the remark this answers is not in the document]
Comment (smacker) [reply Makefile]: in lookout-web it looks like this: https://github.com/src-d/lookout/blob/master/Makefile.web#L69
same-same

[the remark this answers is not in the document]
Comment (dpordomingo) [reply Makefile]: kind of same functionality, but subtle different code (ie. `GO_RUN_ARGS` does not appear in that `Makefile.web`)
But as I said, not blocker at all, so we can merge as it is.
(I also locally tested, and worked like a charm)

[the remark this answers is not in the document]
Comment (smacker) [reply Makefile]: yeah. Maybe we can actually extract go-handler part & Makefile somewhere and then just re-use in our projects. @carlosms wdyt? Shall I create an issue for that?

[the remark this answers is not in the document]
Comment (dpordomingo) [reply Makefile]: +1
I've been proposing that since years :tada:

[the remark this answers is not in the document]
Comment (carlosms) [reply Makefile]: sounds good to me. Let's open that issue :)

File: server/cmd/bblfsh-web/main.go [hunk 2]
```diff
@@ -55,27 +51,12 @@ func main() {
...
-		}
-	}
-	indexPath := filepath.Join(dir, "index.html")
-	r.StaticFile("/", indexPath)
-	r.Static("/static", filepath.Join(dir, "static"))
 	server.Mount(s, r.Group("/api"))
-	// we handle urls on frontend
-	r.NoRoute(func(c *gin.Context) { c.File(indexPath) })
+
+	// TODO(@smacker): add configuration for ServerURL and FooterHTML
```
Comment (carlosms) [inline server/cmd/bblfsh-web/main.go]: This is functionality brought from the code c&p from lookout, but we can't use yet in bblfsh web, right?
Should we have an issue for this, to implement it later?

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (smacker) [reply Makefile]: https://github.com/src-d/feature-idea/issues/129

File: server/cmd/bblfsh-web/main.go (hunk 2 above)
Comment (smacker) [reply server/cmd/bblfsh-web/main.go]: https://github.com/bblfsh/web/issues/201

**Pull request merged**